qemu/tests/blockdev_mirror_stress.py

```diff
@@ -1,24 +1,53 @@
+import six
+import time
+import random
+
 from virttest import utils_test
 
-from provider.blockdev_mirror_parallel import BlockdevMirrorParallelTest
+from provider.storage_benchmark import generate_instance
+from provider.blockdev_mirror_wait import BlockdevMirrorWaitTest
+
 
+class BlockdevMirrorStressTest(BlockdevMirrorWaitTest):
+    """Do block-mirror with fio test as background test"""
 
-class BlockdevMirrorStressTest(BlockdevMirrorParallelTest):
-    """Do block-mirror and vm stress test in parallel"""
+    def fio_thread(self):
+        fio_options = self.params.get("fio_options")
+        if fio_options:
+            self.test.log.info("Start to run fio")
+            self.fio = generate_instance(self.params, self.main_vm, 'fio')
+            fio_run_timeout = self.params.get_numeric("fio_timeout", 2400)
+            self.fio.run(fio_options, fio_run_timeout)
 
-    def stress_test(self):
-        """Run stress testing on vm"""
-        self.stress = utils_test.VMStress(self.main_vm, "stress", self.params)
-        self.stress.load_stress_tool()
+    def remove_files_from_system_image(self, tmo=60):
+        """Remove testing files from system image"""
+        tag_dir_list = [(t, d[1]) for t, d in six.iteritems(self.disks_info) if d[0] == "system"]
+        if tag_dir_list:
+            tag, root_dir = tag_dir_list[0]
+            files = ["%s/%s" % (root_dir, f) for f in self.files_info[tag]]
+            files.append("%s/%s" % (self.params["mnt_on_sys_dsk"], self.params["file_fio"]))
+            rm_cmd = "rm -f %s" % " ".join(files)
 
-    def check_stress_running(self):
-        """stress should be running after block-mirror"""
-        if not self.stress.app_running():
-            self.test.fail("stress stopped unexpectedly")
+            # restart main vm for the original system image is offlined
+            # and the mirror image is attached after block-mirror
+            self.prepare_main_vm()
+            session = self.main_vm.wait_for_login()
+            try:
+                session.cmd(rm_cmd, timeout=tmo)
+            finally:
+                session.close()
 
     def do_test(self):
-        self.blockdev_mirror()
-        self.check_stress_running()
+        bg_test = utils_test.BackgroundTest(self.fio_thread, "")
+        bg_test.start()
```

Comment:
hi aihua, I hit the 'got stuck' issue, I am not sure if it's the same issue as the one we hit when running blockdev_commit_fio, but it's interesting, and I guessed it's a threading issue, so I set the thread as a daemon thread, and did some tests, now it looks like the issue was gone. I cannot look into this issue now, and just suggest that we set a daemon thread here:

bg_test.thread.daemon = True
bg_test.start()

why deamon thread? because the thread quits as soon as the program(fio) quits

Reply:
@zhencliu, thanks for your suggestion, have updated the code, please help to review again.